example/javascriptlib/linting/1-linting/build.mill

```diff
@@ -0,0 +1,56 @@
+// and `prettier` https://prettier.io/docs/en out of the box.
+// You can lint your projects code by providing a configuration for your preferred linter and running `mill _.checkFormatAll`.
+
+// If both configuration files are present, the command `mill _.checkFormatAll` will default to eslint.
+// If neither files are present, the command will exit with an error, you must include at least one configuration file.
+// You can lint via a specificied linter via the commands `mill _.checkFormatEslint` to lint with eslint and
+// `mill _.checkFormatPrettier` to lint with prettier.
+
+// When using prettier you can specify the path to lint via command line argument, `mill _.checkFormatAll "*/**/*.ts"`
+// just as you would when running `prettier --check` if no path is provided mill will default to using "*/**/*.ts".
+// Also if a `.prettierignore` is not provided, mill will generate one ignoring "node_modules" and ".git".
+
+// You can define `npmLintDeps` field to add dependencies specific to linting to your module.
+// The task `npmLintDeps` works the same way as `npmDeps` or `npmDevDeps`.
+
+/** Usage
+> cat foo/src/foo.ts # initial poorly formatted source code
+export class Foo{
+static main(
+args: string[
+])
+{console.log("Hello World!")
+}
+}
+
+> mill foo.checkFormatAll # run linter - since both eslint and prettier configurations are present, mill will opt to use eslint.
+...
+foo/src/foo.ts
+  2:1   error  Expected indentation of 2 spaces but found 0                                   indent
+  3:1   error  Expected indentation of 4 spaces but found 0                                   indent
+  5:1   error  Opening curly brace does not appear on the same line as controlling statement  brace-style
+  5:1   error  Statement inside of curly braces should be on next line                        brace-style
+  5:1   error  Requires a space after '{'                                                     block-spacing
+  5:1   error  Expected indentation of 2 spaces but found 0                                   indent
+  5:14  error  Strings must use singlequote                                                   quotes
+  5:29  error  Missing semicolon                                                              semi
+  6:1   error  Expected indentation of 2 spaces but found 0                                   indent
+  7:2   error  Newline required at end of file but not found                                  eol-last
+...
+...10 problems (10 errors, 0 warnings)
+...10 errors and 0 warnings potentially fixable with running foo.reformatAll.
+
+> rm -rf eslint.config.mjs # since there is no eslint config file `eslint.config.(js|mjs|cjs)`, mill will use prettier if a `.prettierrc` conf file is available.
+
+> mill foo.checkFormatAll # run lint with prettier configuration.
+Checking formatting...
+[warn] foo/src/foo.ts
+[warn] Code style issues found. Run foo.reformatAll to fix.
+*/
```

Comment:
This example should focus on non-autoformat-able lint errors, as the `2-autoformatting/` section already covers those. This section should also demonstrate that fixing the error (e.g. using `sed`) makes the lint errors go away.